mongo/src/main/java/com/redhat/lightblue/mongo/metadata/BSONParser.java

```diff
@@ -51,140 +51,86 @@
 import com.redhat.lightblue.util.JsonUtils;
 
 @SuppressWarnings({"rawtypes", "unchecked"})
-public class BSONParser extends MetadataParser<BSONObject> {
+public class BSONParser extends MetadataParser<Object> {
     private static final Logger LOGGER = LoggerFactory.getLogger(BSONParser.class);
 
     public static final String DELIMITER_ID = "|";
 
-    public BSONParser(Extensions<BSONObject> ex,
+    public BSONParser(Extensions ex,
                       TypeResolver resolver) {
         super(ex, resolver);
     }
 
     @Override
-    public String getStringProperty(BSONObject object, String name) {
-        Object x = object.get(name);
-        if (x != null) {
-            if (x instanceof String) {
-                return (String) x;
-            } else {
-                throw Error.get(MetadataConstants.ERR_ILL_FORMED_METADATA, name);
-            }
-        } else {
-            return null;
-        }
-    }
-
-    @Override
-    public BSONObject getObjectProperty(BSONObject object, String name) {
-        Object x = object.get(name);
-        if (x != null) {
-            if (x instanceof BSONObject) {
-                return (BSONObject) x;
-            } else {
-                throw Error.get(MetadataConstants.ERR_ILL_FORMED_METADATA, name);
-            }
-        } else {
-            return null;
-        }
-    }
-
-    @Override
-    public Object getValueProperty(BSONObject object, String name) {
-        Object x = object.get(name);
-        if (x != null) {
-            if (x instanceof Number
-                    || x instanceof String
-                    || x instanceof Date
-                    || x instanceof Boolean) {
-                return x;
-            } else {
-                throw Error.get(MetadataConstants.ERR_ILL_FORMED_METADATA, name);
-            }
-        } else {
-            return null;
-        }
-    }
-
-    @Override
-    public List<String> getStringList(BSONObject object, String name) {
-        Object x = object.get(name);
-        if (x != null) {
-            if (x instanceof List) {
-                ArrayList<String> ret = new ArrayList<>();
-                for (Object o : (List) x) {
-                    ret.add(o.toString());
-                }
-                return ret;
-            } else {
-                throw Error.get(MetadataConstants.ERR_ILL_FORMED_METADATA, name);
-            }
-        } else {
-            return null;
-        }
+    public Object newMap() {
+        return new BasicDBObject();
```

Comment:
I won't block on this, but as a rule of thumb, I try to return Objects at the highest level possible and pass in parameters at the lowest level as possible. In this case, if there were a use case to use BSONParser directly, then casting wouldn't be required.